modules/insticatorBidAdapter.js

```diff
@@ -0,0 +1,272 @@
+  if (typeof deviceConfig === 'object') {
+    Object.assign(device, deviceConfig);
+  }
+
+  return device;
+}
+
+function buildRegs(bidderRequest) {
+  if (bidderRequest.gdprConsent) {
+    return {
+      ext: {
+        gdpr: bidderRequest.gdprConsent.gdprApplies ? 1 : 0,
+        gdprConsentString: bidderRequest.gdprConsent.consentString,
+      },
+    };
+  }
+
+  return {};
+}
+
+function buildUser() {
+  const userId = getUserId() || generateUUID();
+
+  setUserId(userId);
+
+  return {
+    id: userId,
+  };
+}
+
+function buildRequest(validBidRequests, bidderRequest) {
+  const req = {
+    id: bidderRequest.bidderRequestId,
+    tmax: bidderRequest.timeout,
+    source: {
+      fd: 1,
+      tid: bidderRequest.auctionId,
+    },
+    site: {
+      domain: location.hostname,
+      page: location.href,
+      ref: bidderRequest.refererInfo.referer,
+    },
+    device: buildDevice(),
+    regs: buildRegs(bidderRequest),
+    user: buildUser(),
+    imp: validBidRequests.map((bidRequest) => buildImpression(bidRequest)),
+  };
+
+  const params = config.getConfig('insticator.params');
+
+  if (params) {
+    req.ext = {
+      insticator: params,
+    };
+  }
+
+  return req;
+}
+
+function buildBid(bid, bidderRequest) {
+  const originalBid = bidderRequest.bids.find((b) => b.bidId === bid.impid);
+
+  return {
+    requestId: bid.impid,
+    creativeId: bid.crid,
+    cpm: bid.price,
+    currency: 'USD',
+    netRevenue: true,
+    ttl: bid.exp || config.getConfig('insticator.bidTTL') || BID_TTL,
+    width: bid.w,
+    height: bid.h,
+    mediaType: 'banner',
+    ad: bid.adm,
+    adUnitCode: originalBid.adUnitCode,
+  };
+}
+
+function buildBidSet(seatbid, bidderRequest) {
+  return seatbid.bid.map((bid) => buildBid(bid, bidderRequest));
+}
+
+function validateSize(size) {
+  return (
+    size instanceof Array &&
+    size.length === 2 &&
+    typeof size[0] === 'number' &&
+    typeof size[1] === 'number'
+  );
+}
+
+function validateSizes(sizes) {
+  return (
+    sizes instanceof Array &&
+    sizes.length > 0 &&
+    sizes.map(validateSize).reduce((a, b) => a && b, true)
+  );
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  isBidRequestValid: function (bid) {
+    if (!bid.params.adUnitId) {
+      logError('insticator: missing adUnitId bid parameter');
+      return false;
+    }
+
+    if (!(BANNER in bid.mediaTypes)) {
+      logError('insticator: expected banner in mediaTypes');
+      return false;
+    }
+
+    if (
+      !validateSizes(bid.sizes) &&
+      !validateSizes(bid.mediaTypes.banner.sizes)
+    ) {
+      logError('insticator: banner sizes not specified or invalid');
+      return false;
+    }
+
+    return true;
+  },
+
+  buildRequests: function (validBidRequests, bidderRequest) {
+    const requests = [];
+
+    if (validBidRequests.length > 0) {
+      requests.push({
+        method: 'POST',
+        url: config.getConfig('insticator.endpointUrl') || ENDPOINT,
```

Comment:
`insticator.endpointUrl` supplied by the publisher in setConfig can use the `http:` non secure protocol, which violates the Prebid.org requirement that all endpoints URLs be secure.

Can you please add a check for that?